PWGDQ/Tasks/tableReader.cxx

```diff
@@ -351,6 +363,196 @@ struct AnalysisMuonSelection {
+      if (!twoTracksFilterMap) {
+        continue;
+      }
+
+      // pairing
+      VarManager::FillPair<TPairType, TTrackFillMap>(track1, track2);
+      uint8_t track1Untagged = uint8_t(0);
+      uint8_t track2Untagged = uint8_t(0);
+
+      // Fill pair selection map and fill pair histogram
+      for (int icut = 0; icut < nCuts; icut++) {
+        int trackCutId = fTrackCutsId.at(icut);
+        if (!(twoTracksFilterMap & (uint8_t(1) << trackCutId)))
+          continue;
+
+        AnalysisCompositeCut pairCut = fPairCuts.at(icut);
+        if (pairCut.IsSelected(VarManager::fgValues)) {
+          if (fQA) {
+            fHistMan->FillHistClass(Form("PairsDalitz_track%d_%s", trackCutId, pairCut.GetName()), VarManager::fgValues);
+          }
+
+          // Check if tracks were already tagged
+          bool b1 = dalitzmap[track1.globalIndex()] & (uint8_t(1) << icut);
+          if (!b1) {
+            track1Untagged |= (uint8_t(1) << icut);
+            if (fQA) {
+              ((TH1I*)fStatsList->At(0))->Fill(icut);
+            }
+          }
+          bool b2 = dalitzmap[track2.globalIndex()] & (uint8_t(1) << icut);
+          if (!b2) {
+            track2Untagged |= (uint8_t(1) << icut);
+            if (fQA) {
+              ((TH1I*)fStatsList->At(0))->Fill(icut);
+            }
+          }
+        }
+      }
+
+      // Tag tracks which are not already tagged
+      dalitzmap[track1.globalIndex()] |= track1Untagged;
+      dalitzmap[track2.globalIndex()] |= track2Untagged;
+
+      // Fill track histograms if not already tagged
+      if (fQA) {
+        VarManager::FillTrack<TTrackFillMap>(track1);
+        for (int icut = 0; icut < nCuts; icut++) {
+          if (track1Untagged & (uint8_t(1) << icut)) {
+            int trackCutId = fTrackCutsId.at(icut);
+            AnalysisCompositeCut pairCut = fPairCuts.at(icut);
+            fHistMan->FillHistClass(Form("TrackBarrel_track%d_%s", trackCutId, pairCut.GetName()), VarManager::fgValues);
+          }
+        }
+        VarManager::FillTrack<TTrackFillMap>(track2);
+        for (int icut = 0; icut < nCuts; icut++) {
+          if (track2Untagged & (uint8_t(1) << icut)) {
+            int trackCutId = fTrackCutsId.at(icut);
+            AnalysisCompositeCut pairCut = fPairCuts.at(icut);
+            fHistMan->FillHistClass(Form("TrackBarrel_track%d_%s", trackCutId, pairCut.GetName()), VarManager::fgValues);
+          }
+        }
+      }
+    } // end of tracksP,N loop
+  }
+
+  void processBarrelSkimmed(MyEventsSelected const& events, soa::Filtered<MyBarrelTracksSelected> const& filteredTracks, MyBarrelTracks const& tracks)
+  {
+    dalitzmap.clear();
+    for (auto& event : events) {
+      if (event.isEventSelected()) {
+        auto groupedFilteredTracks = filteredTracks.sliceBy(perCollision, event.globalIndex());
+        runDalitzPairing<gkTrackFillMap>(groupedFilteredTracks, groupedFilteredTracks);
```

Comment:
Here you run a double loop using the tracks from the prefilter group. This is not the way we should do this. The pairing should be done between the "good" tracks and the tracks from the prefiltering group, which are usually selected with more open cuts to increase the probability of finding a conversion. 
Also, the prefiltering should have an option of running over same event pairs. This is not typical, however, sometimes we can diagnose various problems in analysis, such as track duplicates, etc.